x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -201,18 +199,12 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 		}
 
 		// pairing is valid, we can pay provider for work
-		coinsPerCu := k.Keeper.MintCoinsPerCU(ctx)
-		reward := coinsPerCu.MulInt64(int64(rewardedCU))
-		if reward.IsZero() {
-			continue
-		}
-
-		rewardCoins := sdk.Coins{sdk.Coin{Denom: epochstoragetypes.TokenDenom, Amount: reward.TruncateInt()}}
+		rewardedCUDec := sdk.OneDec().MulInt64(int64(rewardedCU))
```

Comment:
why do we need this calculation? just make the dec with rewardedCU

Reply:
done. see https://github.com/lavanet/lava/commit/36c4bd192408759547b2588605992db3d3b23f21